pkg/controller/drain/drain_controller.go

```diff
@@ -343,6 +335,59 @@ func (ctrl *Controller) syncNode(key string) error {
 	return nil
 }
 
+func (ctrl *Controller) drainNode(node *corev1.Node, drainer *drain.Helper) error {
+	isOngoingDrain := false
+	var duration time.Duration
+
+	for k, v := range ctrl.ongoingDrains {
+		if k != node.Name {
+			continue
+		}
+		isOngoingDrain = true
+		duration = time.Now().Sub(v)
+		glog.Infof("Previous node drain found. Drain has been going on for %v hours", duration.Hours())
+		if duration > ctrl.cfg.DrainTimeoutDuration {
+			glog.Errorf("node %s: drain exceeded timeout: %v. Will continue to retry.", node.Name, ctrl.cfg.DrainTimeoutDuration)
+			ctrlcommon.MCCDrainErr.Set(1)
+		}
+		break
+	}
+
+	if !isOngoingDrain {
+		ctrl.logNode(node, "cordoning")
+		// perform cordon
+		if err := ctrl.cordonOrUncordonNode(true, node, drainer); err != nil {
+			return fmt.Errorf("node %s: failed to cordon: %w", node.Name, err)
+		}
+		ctrl.ongoingDrains[node.Name] = time.Now()
+	}
+
+	// Attempt drain
+	ctrl.logNode(node, "initiating drain")
+	if err := drain.RunNodeDrain(drainer, node.Name); err != nil {
+		// To mimic our old daemon logic, we should probably have a more nuanced backoff.
+		// However since the controller is processing all drains, it is less deterministic how soon the next drain will retry,
+		// Anywhere between instant (if a node change happened) or up to hours (if there are many nodes competing for resources)
+		// For now, let's say if a node has been trying for a set amount of time, we make it less prioritized.
+		if duration > ctrl.cfg.DrainRequeueFailingThreshold {
+			ctrl.logNode(node, "Drain failed. Drain has been failing for more than %v minutes. Waiting %v minutes then retrying. "+
+				"Error message from drain: %v", ctrl.cfg.DrainRequeueFailingThreshold.Minutes(), ctrl.cfg.DrainRequeueFailingDelay.Minutes(), err)
+			ctrl.enqueueAfter(node, ctrl.cfg.DrainRequeueFailingDelay)
+			ctrlcommon.MCCDrainErr.Set(1)
```

Comment:
So we have this also set to 10 minutes (as a slowdown), but not as a hard fail. To match existing behaviour, what do you think about not alerting here, and only alert on the above 1hr timeout duration?